web_widget_x2many_2d_matrix/static/src/js/2d_matrix_renderer.js

```diff
@@ -2,422 +2,537 @@
+        },
+
+        /**
+         * Wraps the column aggregate with a tfoot element
+         *
+         * @private
+         * @returns {jQueryElement} The footer element with the cells in it.
+         */
+        _renderFooter: function () {
+            var $cells = this._renderAggregateColCells();
+            if ($cells) {
+                return $('<tfoot>').append(
+                    $('<tr>').append('<td/>').append($cells)
+                );
+            }
+        },
+
+        /**
+         * Render the Aggregate cells for the column.
+         *
+         * @private
+         * @returns {List} the rendered cells
+         */
+        _renderAggregateColCells: function () {
+            var self = this;
+            return _.map(this.columns, function (column) {
+                var $cell = $('<td>', {class: 'col-total text-right'});
+                if (column.aggregate) {
+                    self._apply_aggregate_value($cell, column.aggregate);
+                }
+                return $cell;
+            });
+        },
+
+        /**
+         * Compute the column aggregates.
+         * This function is called everytime the value is changed.
+         *
+         * @private
+         */
+        _computeColumnAggregates: function () {
+            if (!this.matrix_data.show_column_totals) {
+                return;
+            }
+            var fname = this.matrix_data.field_value,
+                field = this.state.fields[fname];
+            if (!field) {
+                return;
+            }
+            var type = field.type;
+            if (!~['integer', 'float', 'monetary'].indexOf(type)) {
+                return;
+            }
+            _.each(this.columns, function (column, index) {
+                column.aggregate = {
+                    fname: fname,
+                    ftype: type,
+                    help: _t('Sum'),
+                    value: 0,
+                };
+                _.each(this.rows, function (row) {
+                    column.aggregate.value += row.data[index].data[fname];
+                });
+            });
+        },
+
+        /**
+         * @override
+         */
+        updateState: function (state, params) {
+            if (params.matrix_data) {
+                this._saveMatrixData(params.matrix_data);
+            }
+            return this._super.apply(this, arguments);
+        },
+
+        /**
+        * Traverse the fields matrix with the keyboard
+        *
+        * @override
+        * @private
+        * @param {OdooEvent} event "navigation_move" event
+        */
+        _onNavigationMove: function (event) {
+            var widgets = this.__parentedChildren,
+                index = widgets.indexOf(event.target),
+                first = index === 0,
+                last = index === widgets.length - 1,
+                move = 0;
+            // Guess if we have to move the focus
+            if (event.data.direction === "next" && !last) {
```

Comment:
can't we support also direction="down|up" to jump on the next/prev matrix line?

Reply:
I added it to roadmap.